### PR TITLE
Make collection dashboard question buttons consistent

### PR DESCRIPTION
**Pull request by WiNloSt**
Epic #22826
Task: https://www.notion.so/metabase/Inconsistent-hover-effects-on-action-buttons-5e619a2f52e04765872e82f424260bdd

#### Note
- This is a follow-up work from https://github.com/metabase/metabase/pull/23833
- After https://github.com/metabase/metabase/pull/23833 is merged, this PR should change the base branch back to `master` now it's based on that PR to accommodate the PR review.

This PR:
- Change collection, dashboard, and question header [button icon size to `16px`](https://metaboat.slack.com/archives/C03E8MQJZBM/p1657818022315299?thread_ts=1657814825.802049&cid=C03E8MQJZBM)
- Make them all have a margin of `0.5rem`
- Make them have the same hover effect (background color, text color)

### CODE REVIEW

File: frontend/src/metabase/collections/components/CollectionHeader/CollectionHeader.styled.tsx [hunk 1]
```diff
@@ -37,6 +41,10 @@ export const CollectionHeaderButton = styled(
     color: ${color("brand")};
     background-color: ${color("bg-medium")};
   }
+
+  ${Button.Content} {
+    height: 100%;
+  }
```
Comment (WiNloSt) [inline frontend/src/metabase/collections/components/CollectionHeader/CollectionHeader.styled.tsx:47]: since we're shrinking the icon size, now the icon stays at the top, not center, so setting height to 100% fixes this since the button content is a flex container that center align its children.

File: frontend/src/metabase/dashboard/containers/DashboardHeader.jsx [hunk 2]
```diff
@@ -255,17 +254,12 @@ class DashboardHeader extends Component {
           >
             <div>
               <Tooltip tooltip={t`Add a filter`}>
-                <a
+                <DashboardHeaderButton
```
Comment (WiNloSt) [inline frontend/src/metabase/dashboard/containers/DashboardHeader.jsx:257]: Prevent having a `button` inside an `a` tag.

File: frontend/src/metabase/dashboard/containers/DashboardHeader.jsx [hunk 3]
```diff
@@ -350,12 +344,13 @@ class DashboardHeader extends Component {
               }
             />
           </Tooltip>,
-          <EntityMenu
-            key="dashboard-action-menu-button"
-            items={extraButtons}
-            triggerIcon="ellipsis"
-            tooltip={t`Move, archive, and more...`}
-          />,
+          <DashboardHeaderButtonContainer key="dashboard-action-menu-button">
```
Comment (WiNloSt) [inline frontend/src/metabase/dashboard/containers/DashboardHeader.jsx:347]: Some buttons are wrapped with a container since it's not possible to style the button itself due to their implementations and usages.

File: frontend/src/metabase/query_builder/components/QuestionActions.styled.tsx [hunk 4]
```diff
@@ -21,32 +13,3 @@ export const StrengthIndicator = styled(DatasetMetadataStrengthIndicator)`
   float: none;
   margin-left: 3.5rem;
 `;
-
-export type AnimationStates = "expand" | "shrink" | null;
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/QuestionActions.styled.tsx:25]: @npfitz Sorry to delete your implementation here, but since `BookmarkToggle` is already present, I remove this in favor of reusing a component instead. Let me know if there's any concern as I might not know all of the context why we need a separate bookmark toggle in the first place.

File: frontend/src/metabase/query_builder/components/view/ViewHeader.jsx [hunk 5]
```diff
@@ -460,43 +461,51 @@ function ViewTitleHeaderRightSide(props) {
...
-          onCancel={cancelQuery}
-        />
+        <ViewHeaderIconButtonContainer>
+          <RunButtonWithTooltip
+            className={cx("text-brand-hover text-dark", {
+              hide: isShowingNotebook,
+              "text-white-hover": isResultDirty,
+            })}
+            iconSize={16}
+            onlyIcon
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/view/ViewHeader.jsx:497]: change from `borderless` to `onlyIcon` as `borderless` still render border but with `transparent` color causing the size difference when changing between active and inactive button.

https://github.com/metabase/metabase/blob/13de0b8327c1250e74fd321234f216b502ae5726/frontend/src/metabase/css/components/buttons.css#L110-L114
vs
https://github.com/metabase/metabase/blob/13de0b8327c1250e74fd321234f216b502ae5726/frontend/src/metabase/css/components/buttons.css#L120-L125

**Review by npfitz — COMMENTED** [review]
The Events Icon on the Collections header looks like it might be the wrong size: 
image

**Review by npfitz — APPROVED** [review]
Aside from the event icon, this looks good to me. All buttons seem to have a consistent hover effect

**Comment by WiNloSt** [comment]
> The Events Icon on the Collections header looks like it might be the wrong size:

@npfitz How do you reproduce that?

I tried 3 different browsers and couldn't see that

#### Safari
image

#### Firefox
image

#### Chrome
image

One thing may be worth pointing out though. It seems something's up with our FE build lately. I reviewed @iethree 's PR a few times yesterday and switching back and forth between branches seems to break the build or the CSS I'm not sure I haven't dug deep enough, but I need to bust the webpack cache and build FE again every time. But I know the style isn't updated as the bulk filter was definitely not the latest version (no icons) every time I switch back to the bulk filter's branch. @iethree Have you experienced that too? Also, @ranquild [seems you found](https://github.com/metabase/metabase/pull/23941#pullrequestreview-1038501825) that when reviewing my attempted fix for the dashboard parameters header too right?

In short, I'm not really sure if this problem occurred recently, but from my experience within the last few days, it feels like so.

**Comment by npfitz** [comment]
@WiNloSt my apologies, I must not have fetched latest before testing. Did a fetch and ran things this morning and now I'm getting properly sized Icons. 🎉

File: frontend/src/metabase/collections/components/CollectionHeader/CollectionHeader.styled.tsx [hunk 6]
```diff
@@ -21,6 +21,7 @@ export const HeaderActions = styled.div`
   display: flex;
   margin-top: 0.5rem;
   align-self: start;
+  gap: 0.5rem;
```
Comment (WiNloSt) [inline frontend/src/metabase/collections/components/CollectionHeader/CollectionHeader.styled.tsx:24]: Use gap the same as https://github.com/metabase/metabase/pull/23833/commits/bf2e2d6625657aee776ccdaab61c0d9c04f1b289#diff-13a900a69321a6b06dbe132e97c09e50cc55b358de6031e32844ecb04eeccbc0R158

File: frontend/src/metabase/collections/components/CollectionHeader/CollectionHeader.styled.tsx [hunk 7]
```diff
@@ -30,13 +31,18 @@ interface CollectionHeaderButtonProps {
 export const CollectionHeaderButton = styled(
   Button,
 )<CollectionHeaderButtonProps>`
-  padding: 0.5rem 0.75rem;
-  height: 2.5rem;
+  padding: 0.25rem 0.5rem;
```
Comment (WiNloSt) [inline frontend/src/metabase/collections/components/CollectionHeader/CollectionHeader.styled.tsx:34]: Use the same value as Maz https://github.com/metabase/metabase/pull/23833/commits/bf2e2d6625657aee776ccdaab61c0d9c04f1b289#diff-5a30b16741703579e402851f93daa3f5b97928cd3d13e0fade912be23c5d2940R39-R41

File: frontend/src/metabase/query_builder/components/view/ViewHeader.styled.jsx [hunk 8]
```diff
@@ -197,8 +198,17 @@ export const ViewHeaderActionPanel = styled.div`
     margin-top: 1rem;
     padding: 0.5rem 2.5rem 0 2rem;
   }
+`;
 
+export const ViewHeaderIconButtonContainer = styled.div`
   ${Button.Root} {
-    margin-left: 0.5rem;
+    padding: 0.25rem 0.5rem;
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/view/ViewHeader.styled.jsx:205]: Use same spacing as Maz https://github.com/metabase/metabase/pull/23833/commits/bf2e2d6625657aee776ccdaab61c0d9c04f1b289#diff-5a30b16741703579e402851f93daa3f5b97928cd3d13e0fade912be23c5d2940R39-R41

File: frontend/src/metabase/dashboard/containers/DashboardHeader.styled.jsx [hunk 9]
```diff
@@ -5,8 +5,8 @@ import Button from "metabase/core/components/Button";
 
 export const DashboardHeaderActionDivider = styled.div`
   height: 1.25rem;
-  padding-left: 0.75rem;
-  margin-left: 0.75rem;
+  padding-left: 0.5rem;
```
Comment (WiNloSt) [inline frontend/src/metabase/dashboard/containers/DashboardHeader.styled.jsx:8]: To match `QuestionActionsDivider`'s margin

https://github.com/metabase/metabase/pull/23967/files#diff-bf5deac9a0c26dd2003e0b7fd4682e4cca2fdedab92aba36f3b52f008edb5346R7-R8

**Review by mazameli — APPROVED** [review]
<3
